source/common/http/conn_manager_impl.cc

```diff
@@ -887,6 +887,8 @@ bool ConnectionManagerImpl::ActiveStream::validateHeaders() {
         grpc_status = Grpc::Status::WellKnownGrpcStatus::Internal;
       }
 
+      // We have to make sure that the stream is ready to be closed.
+      maybeEndDecode(end_stream);
```

Comment:
I think we should just move the maybeendDecode in decodeHeaders to before validateHeaders.